specs/agents/transport.md

```diff
@@ -47,4 +47,7 @@ The APM Server accepts both uncompressed and compressed HTTP requests. The follo
 - zlib data format (`Content-Encoding: deflate`)
 - gzip data format (`Content-Encoding: gzip`)
 
-Agents should compress the HTTP payload by default, optimising for speed over compactness (typically known as the "best speed" level).
+Agents MUST compress the HTTP payload, optimising for speed over compactness (typically known as the "best speed" level).
+If the host part of the APM Server URL is either `localhost`, `127.0.0.1`, `::1`, or `0:0:0:0:0:0:0:1`, agents SHOULD disable compression.
+To do so, agents can either completely omit calling the compression library and remove the `Content-Encoding` header,
+or simply use the compresion level `NO_COMPRESSION`.
```

Comment:
This could become an issue when https://github.com/elastic/apm-aws-lambda/issues/71 is implemented. The extension would potentially perceive the data as being compressed as the `Content-Encoding` is `deflate`. This would lead to the data being sent to APM Server uncompressed.

Reply:
Where "This" is `simply use the compresion level NO_COMPRESSION`? Ah, right. I had to re-familiarize myself here. With `NO_COMPRESSION` there will still be the compression header and footer bytes (at least with gzip), so the `Content-Encoding` header must still be specified.

Perhaps expressly disallow using the `NO_COMPRESSION` option for this reason?

Reply:
I'd mentioned yesterday that turning off compression in the Node.js APM agent might be a pain. Don't worry about that for this spec. I can replace the compression stream with a "PassThrough" stream, which should be pretty straightforward.

Reply:
I'm leaning towards explicitly disallowing `NO_COMPRESSION` for that reason.

@trentm, you mentioned this would make it tricky to implement it in the Node.js agent. Any thoughts?
Note that this is an optional part of the spec (SHOULD).